content/articles/wavelet-transform-based-denoising-of-images-using-matlab/index.md

```diff
@@ -0,0 +1,141 @@
+
+![Denoised image](/engineering-education/wavelet-transform-based-denoising-of-images/image5.png)
+We get the denoised image as shown above. This is how you can use the wavelet analyzer tool to the denoise image.
+
+### Functions used in Wavelet Denoising
+`wdencmp` - It can be used for denoising of both 1-D and 2-D signals. This means it can be used for both audios as well as images.
+`syntax`
+```Matlab
+imgden = wdencmp('gbl_or_Ivd',img, 'wname', N, THR, SORH, KEEPAPP)
+```
+where;
+img: is the noisy image that we input.
+imgden: is the output denoised image.
+gbl_or_ivd: either use `gbl` for single threshold or `Ivd` for level-dependent threshold(either use `gbl` or `Ivd`).
+wname: is the wavelet type used.
+N: represents the decomposition levels.
+THR: threshold levels[single value for `gbl` and 3xN matric to `ivd`, three rows each for horizontal, diagonal, and vertical detailed
+co-efficient and N number of columns, where N is the decomposition level].
+SORH: `s` or `h` corresponds to the soft and hard thresholding respectively.
+KEEPAPP: it is either 0 or 1. It is 0 when the approximation coefficients cannot be thresholded and 1 if they can be.
+
+### Functions used for finding the dafault values for denoising(THR, SORH, KEEPAPP)
+The default values are found automatically using `ddencmp`. The function syntax is;
+```matlab
+[THR, SORH, KEEPAPP] = ddencmp('den', 'wv', img)
+```
+
+We pass the input noisy image(img), defined denoising method, and wavelet(WV). Executing this function we get the values for these constants.
+
+### Example code to show use of `wdencmp` and `ddencmp`
+We first load original image. This is done by executing the code below which allows us to select the image from the pc.
+```matlab
+[filename, pathname] = uigetfile('*.*', 'select input image');
+fileWithPath = strcat(pathname, filename);  
+img = imread(fileWithPath); % assigning the image to this variable fileWithPath.
+```
+T read the image, we use the `imread` function. The image `img` is the unnoisy original image, and so we add noise to make it noisy.
+```Matlab
+imgn = imnoise(img, 'gaussian', 0, 0.005);  %add noise
+```
+> Note that we add `gaussian` noise of mean value of 0 and variance is 0.005.
+
+We then define the signal extension mode `dwtmode` which is a periodic extension of the signals. This done by the code below;
+```Matlab
+dwtmode('per'); %signal extension mode
+```
+We first execute `ddencmp` function. So when we execute this function, we will get the `THR`, `SORH`, `KEEPAPP` that will be used in the `wdencmp` function.
+```matlab
+[thr, sorh, keepapp] = ddencmp('den', 'wv', imgn); %finding default values
+```
+After defining the `thr`, `sorh`, `keepapp` as shown below. We pass them to the `wdencmp` function as shown below. This is to denoise images 
+using image thresholding.
+```Matlab
+% Denoise image using global thresholding.
+imgden = wdencmp('gbl', double(imgn), 'sym4', 2, thr, sorh, keepapp);
+```
```

Comment:
All these variable names looks too confusing - since they are shorthands. Please change the names, and elaborate about each.